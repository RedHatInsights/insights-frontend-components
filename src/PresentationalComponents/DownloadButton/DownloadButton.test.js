import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DownloadButton from './DownloadButton';

describe('DownloadButton component', () => {
    describe('should render', () => {
        it('CSV and JSON by default', () => {
            const wrapper = shallow(<DownloadButton />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('custom items', () => {
            const wrapper = shallow(<DownloadButton extraItems={['something']} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('API', () => {
        it('clicking should open dropdown', () => {
            const wrapper = mount(<DownloadButton extraItems={['something']} />);
            wrapper.find('button').first().simulate('click');
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('onSelect should be called with CSV', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={['something']} onSelect={onSelect} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').first().simulate('click');
            expect(onSelect.mock.calls.length).toBe(1);
            expect(onSelect.mock.calls[0][1]).toBe('csv');
        });

        it('onSelect should be called with JSON', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={['something']} onSelect={onSelect} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').at(1).simulate('click');
            expect(onSelect.mock.calls.length).toBe(1);
            expect(onSelect.mock.calls[0][1]).toBe('json');
        });

        it('shouldn\'t call onSelect', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={['something']} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').at(1).simulate('click');
            expect(onSelect.mock.calls.length).toBe(0);
        })
    });
});
