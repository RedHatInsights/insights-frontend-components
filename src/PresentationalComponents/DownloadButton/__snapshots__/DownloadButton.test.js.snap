// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton component API clicking should open dropdown 1`] = `
<DownloadButton
  extraItems={
    Array [
      "something",
    ]
  }
  onSelect={[Function]}
>
  <Dropdown
    className=""
    direction="down"
    dropdownItems={
      Array [
        <Item
          className=""
          component="button"
          href="#"
          isDisabled={false}
          isHovered={false}
          onClick={[Function]}
        >
          CSV
        </Item>,
        <Item
          className=""
          component="button"
          href="#"
          isDisabled={false}
          isHovered={false}
          onClick={[Function]}
        >
          JSON
        </Item>,
        "something",
      ]
    }
    isOpen={true}
    isPlain={true}
    onSelect={[Function]}
    position="left"
    toggle={
      <DropdownToggle
        className=""
        iconComponent={[Function]}
        id=""
        isActive={false}
        isFocused={false}
        isHovered={false}
        isOpen={false}
        onToggle={[Function]}
        parentRef={null}
      >
        <DownloadIcon
          color="currentColor"
          size="lg"
          title={null}
        />
      </DropdownToggle>
    }
  >
    <div
      className="pf-c-dropdown pf-m-plain pf-m-expanded"
    >
      <DropdownToggle
        className=""
        iconComponent={[Function]}
        id="pf-toggle-id-1"
        isActive={false}
        isFocused={false}
        isHovered={false}
        isOpen={true}
        key=".0"
        onToggle={[Function]}
        parentRef={
          <div
            class="pf-c-dropdown pf-m-plain pf-m-expanded"
          >
            <button
              aria-expanded="true"
              class="pf-c-dropdown__toggle"
              id="pf-toggle-id-1"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="2em"
                role="img"
                viewBox="0 0 512 512"
                width="2em"
              >
                <path
                  d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                  transform=""
                />
              </svg>
              <svg
                aria-hidden="true"
                class="pf-c-dropdown__toggle-icon"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  transform=""
                />
              </svg>
            </button>
            <div>
              <ul
                aria-labelledby="pf-toggle-id-1"
                class="pf-c-dropdown__menu"
              >
                <li>
                  <button
                    class="pf-c-dropdown__menu-item"
                    href="#"
                  >
                    CSV
                  </button>
                </li>
                <li>
                  <button
                    class="pf-c-dropdown__menu-item"
                    href="#"
                  >
                    JSON
                  </button>
                </li>
                something
              </ul>
            </div>
          </div>
        }
      >
        <DropdownToggle
          className=""
          id="pf-toggle-id-1"
          isActive={false}
          isFocused={false}
          isHovered={false}
          isOpen={true}
          onToggle={[Function]}
          parentRef={
            <div
              class="pf-c-dropdown pf-m-plain pf-m-expanded"
            >
              <button
                aria-expanded="true"
                class="pf-c-dropdown__toggle"
                id="pf-toggle-id-1"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="2em"
                  role="img"
                  viewBox="0 0 512 512"
                  width="2em"
                >
                  <path
                    d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                    transform=""
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="pf-c-dropdown__toggle-icon"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    transform=""
                  />
                </svg>
              </button>
              <div>
                <ul
                  aria-labelledby="pf-toggle-id-1"
                  class="pf-c-dropdown__menu"
                >
                  <li>
                    <button
                      class="pf-c-dropdown__menu-item"
                      href="#"
                    >
                      CSV
                    </button>
                  </li>
                  <li>
                    <button
                      class="pf-c-dropdown__menu-item"
                      href="#"
                    >
                      JSON
                    </button>
                  </li>
                  something
                </ul>
              </div>
            </div>
          }
        >
          <button
            aria-expanded={true}
            className="pf-c-dropdown__toggle"
            id="pf-toggle-id-1"
            onClick={[Function]}
          >
            <DownloadIcon
              color="currentColor"
              size="lg"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="2em"
                role="img"
                viewBox="0 0 512 512"
                width="2em"
              >
                <path
                  d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                  transform=""
                />
              </svg>
            </DownloadIcon>
            <CaretDownIcon
              className="pf-c-dropdown__toggle-icon"
              color="currentColor"
              size="sm"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="pf-c-dropdown__toggle-icon"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  transform=""
                />
              </svg>
            </CaretDownIcon>
          </button>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownMenu
        aria-labelledby="pf-toggle-id-1"
        className=""
        component="ul"
        isOpen={true}
        onClick={[Function]}
        position="left"
      >
        <FocusTrap
          _createFocusTrap={[Function]}
          active={true}
          focusTrapOptions={Object {}}
          paused={false}
          tag="div"
        >
          <div>
            <ul
              aria-labelledby="pf-toggle-id-1"
              className="pf-c-dropdown__menu"
              hidden={false}
              onClick={[Function]}
            >
              <Item
                className=""
                component="button"
                href="#"
                isDisabled={false}
                isHovered={false}
                key="download-csv"
                onClick={[Function]}
              >
                <DropdownItem
                  className="pf-c-dropdown__menu-item"
                  component="button"
                  href="#"
                  isDisabled={false}
                  isHovered={false}
                  onClick={[Function]}
                >
                  <li>
                    <button
                      className="pf-c-dropdown__menu-item"
                      disabled={false}
                      href="#"
                      onClick={[Function]}
                    >
                      CSV
                    </button>
                  </li>
                </DropdownItem>
              </Item>
              <Item
                className=""
                component="button"
                href="#"
                isDisabled={false}
                isHovered={false}
                key="download-json"
                onClick={[Function]}
              >
                <DropdownItem
                  className="pf-c-dropdown__menu-item"
                  component="button"
                  href="#"
                  isDisabled={false}
                  isHovered={false}
                  onClick={[Function]}
                >
                  <li>
                    <button
                      className="pf-c-dropdown__menu-item"
                      disabled={false}
                      href="#"
                      onClick={[Function]}
                    >
                      JSON
                    </button>
                  </li>
                </DropdownItem>
              </Item>
              something
            </ul>
          </div>
        </FocusTrap>
      </DropdownMenu>
    </div>
  </Dropdown>
</DownloadButton>
`;

exports[`DownloadButton component should render CSV and JSON by default 1`] = `
<Dropdown
  className=""
  direction="down"
  dropdownItems={
    Array [
      <Item
        className=""
        component="button"
        href="#"
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
      >
        CSV
      </Item>,
      <Item
        className=""
        component="button"
        href="#"
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
      >
        JSON
      </Item>,
    ]
  }
  isOpen={false}
  isPlain={true}
  onSelect={[Function]}
  position="left"
  toggle={
    <DropdownToggle
      className=""
      iconComponent={[Function]}
      id=""
      isActive={false}
      isFocused={false}
      isHovered={false}
      isOpen={false}
      onToggle={[Function]}
      parentRef={null}
    >
      <DownloadIcon
        color="currentColor"
        size="lg"
        title={null}
      />
    </DropdownToggle>
  }
/>
`;

exports[`DownloadButton component should render custom items 1`] = `
<Dropdown
  className=""
  direction="down"
  dropdownItems={
    Array [
      <Item
        className=""
        component="button"
        href="#"
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
      >
        CSV
      </Item>,
      <Item
        className=""
        component="button"
        href="#"
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
      >
        JSON
      </Item>,
      "something",
    ]
  }
  isOpen={false}
  isPlain={true}
  onSelect={[Function]}
  position="left"
  toggle={
    <DropdownToggle
      className=""
      iconComponent={[Function]}
      id=""
      isActive={false}
      isFocused={false}
      isHovered={false}
      isOpen={false}
      onToggle={[Function]}
      parentRef={null}
    >
      <DownloadIcon
        color="currentColor"
        size="lg"
        title={null}
      />
    </DropdownToggle>
  }
/>
`;
